### ISSUE TITLE
Support indirect paths to ALTCACHE

### ISSUE THREAD

**Issue by vanrein** [arpa2/mkhere#23]
It seems that `./mkhere/xxx.sh` with an `$ALTCACHE` do not resolve to the actual directory.  This was observed with in-place calls, so with a `.` prefixing the path.  Canonicalisation of the path is then useful.

One sign of this was a negative string length for the prefix,

```
root@2fa99bbdf568:/msys64/home/mkhere# ./arpa2cm.sh build
./lib/stdlib: line 37: $PREFIXLEN: substring expression < 0
002b:fixme:winsock:WS_EnterSingleProtocolW unknown Protocol <0x00000000>
002b:fixme:winsock:WS_EnterSingleProtocolW unknown Protocol <0x00000000>
CMake Error: The source directory "/msys64/home/src/arpa2cm-v0.8.0/arpa2cm-v0.8.0.git" does not exist.
Specify --help for usage, or press the help button on the CMake GUI.
mingw32-make: *** No targets specified and no makefile found.  Stop.
arpa2cm build failed with exit code 2
```

The holding code is

```
TRAILER="${ALTCACHE}/mkhere/${TARGET}.sh" 
PREFIXLEN=$[${#0}-${#TRAILER}] 
NEWROOT="${0:0:$PREFIXLEN}" 
debug "Comparing \"${0:$PREFIXLEN}\" to \"${TRAILER}\"" 
```

Workaround is to canonicalise before calling, so not to use `./mkhere/xxx.sh` but `/path/to/altroot/mkhere/xxx.sh`.  But we can do better, I'm sure.

**Comment by vanrein** [comment]
I suppose `./mkhere/xxx.sh` is not a problem without the ALTCACHE construct, because it would end up doing `chroot .` which would be correct.

However, with ALTCACHE, a comparison is made to the path name, and that will fail.  Canonicalising the name should help.  That's easy with `readlink -f`.

**Comment by vanrein** [comment]
Resolved in 3e0076bc8e3347a82308e98ed101c0f01959d56d in branch `master`.
